backend/api/Areas/Leases/Controllers/LeaseTermController.cs

```diff
@@ -0,0 +1,96 @@
+using MapsterMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Pims.Api.Policies;
+using Pims.Dal.Entities;
+using Pims.Dal.Security;
+using Pims.Dal.Services;
+using Swashbuckle.AspNetCore.Annotations;
+using System.Collections.Generic;
+
+namespace Pims.Api.Areas.Lease.Controllers
+{
+    /// <summary>
+    /// LeaseTermController class, provides endpoints for interacting with lease property improvements.
+    /// </summary>
+    [Authorize]
+    [ApiController]
+    [ApiVersion("1.0")]
+    [Area("leases")]
+    [Route("v{version:apiVersion}/[area]")]
+    [Route("[area]")]
+    public class LeaseTermController : ControllerBase
+    {
+        #region Variables
+        private readonly IPimsService _pimsService;
+        private readonly IMapper _mapper;
+        #endregion
+
+        #region Constructors
+        /// <summary>
+        /// Creates a new instance of a LeaseTermController class, initializes it with the specified arguments.
+        /// </summary>
+        /// <param name="pimsService"></param>
+        /// <param name="mapper"></param>
+        ///
+        public LeaseTermController(IPimsService pimsService, IMapper mapper)
+        {
+            _pimsService = pimsService;
+            _mapper = mapper;
+        }
+        #endregion
+
+        #region Endpoints
+        /// <summary>
+        /// Update the specified term on the passed lease.
+        /// </summary>
+        /// <returns></returns>
+        [HttpPost("{leaseId:long}/term")]
+        [HasPermission(Permissions.LeaseEdit)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(IEnumerable<Models.Lease.LeaseModel>), 200)]
+        [SwaggerOperation(Tags = new[] { "lease" })]
+        public IActionResult AddTerm(long leaseId, Models.Lease.TermModel termModel)
```

Comment:
Not sure if this would be a big deal in C# / ASP but shouldn't we annotate the model as coming in the POST body via `[FromBody]` ?

Reply:
good point, I'll add that in.